src/Lists.js

```diff
@@ -0,0 +1,264 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { useMediaQuery } from 'react-responsive';
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import { initializeApp } from 'firebase/app';
+import { getFirestore } from "firebase/firestore";
+import { serverTimestamp } from "firebase/firestore";
+import { setDoc, doc } from "firebase/firestore";
+import { query, collection } from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+import Tab from '@mui/material/Tab';
+import Tabs from '@mui/material/Tabs';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MenuIcon from '@mui/icons-material/Menu';
+import IconButton from '@mui/material/IconButton';
+import List from './List';
+
+// web app's Firebase configuration
+const firebaseConfig = {
+    apiKey: "AIzaSyDnLHzrDzijh4KmeJWRU5zSIiW2cPsZRHU",
+    authDomain: "cs124-lab3-3028f.firebaseapp.com",
+    projectId: "cs124-lab3-3028f",
+    storageBucket: "cs124-lab3-3028f.appspot.com",
+    messagingSenderId: "426502461839",
+    appId: "1:426502461839:web:56b4c42f33c8a6187353fa"
+};
+    
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
+
+export default function Lists(props) {
+    const q = query(collection(db, 'Lists'));   
+    const [lists, loading, error] = useCollectionData(q);
+    const [tabId, setTabId] = useState((lists && lists.length!==0) ? lists[0].id : 'none');
```

Comment:
Since loading is going to be true (and lists undefined) the first time this function is called, useState is always going to be initialized to 'none'.  So, just do useState('none')

Reply:
On next line, you can: if (tabId === 'none' && lists && lists.length != 0) { tabId = lists[0].id}